### PR TITLE
Add KDB compression

### PR DESCRIPTION
**Pull request by Kinoye**
Add function kdbzip using .cmp.compression; Also added a functionality which list all compressed files in a table

### CODE REVIEW

File: code/processes/housekeeping.q [hunk 1]
```diff
@@ -60,6 +73,18 @@ wrapper:{[DICT]
 
 //FUNCTIONS FOR LINUX
 
+
+
+//-compress by calling .cmp.compress function defined
+kdbzip:{[FILE]   
+	 @[{.lg.o[`housekeeping;"compressing ",x]; .cmp.compress[hsym `$x ;2;17;4;hcount hsym `$x]};
+	     FILE; 
+		 {.lg.e[`housekeeping;"Failed to compress ",x," : ", y]}[FILE]]
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: Can remove the square brackets around the last `FILE`.

File: code/processes/housekeeping.q [hunk 2]
```diff
@@ -60,6 +73,18 @@ wrapper:{[DICT]
 
 //FUNCTIONS FOR LINUX
 
+
+
+//-compress by calling .cmp.compress function defined
+kdbzip:{[FILE]   
+	 @[{.lg.o[`housekeeping;"compressing ",x]; .cmp.compress[hsym `$x ;2;17;4;hcount hsym `$x]};
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: ```hsym`$x``` is used twice, could save as a temp var and call that + can remove some whitespace.

File: code/processes/housekeeping.q [hunk 3]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
 		//-if correctly columned csv has nulls, report error and skip lines 
 		[if[(any nullcheck:any null (housekeepingcsv.function;housekeepingcsv.age))>0; .lg.o[`housekeeping;"Null values found in file, skipping line(s)  ", ("," sv (string where nullcheck))]];
 		housekeepingcsv2:(housekeepingcsv[where not nullcheck]);
-			wrapper each housekeepingcsv2]]}
+			wrapper each housekeepingcsv2]];
+		
+	 //-list all the compressed file and save in a table
+	hdbfilepaths: filepaths where filepaths like "*database20*";
+	files: asc last each ("/" vs' hdbfilepaths);
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: Can remove the parentheses and remove some whitespace.

File: code/processes/housekeeping.q [hunk 4]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
...
+    filehandlesall: asc hsym `$hdbfilepaths;
+	compressedfiles: files where 0 < count each {-21!x} each filehandlesall;
+	compressedfilehandles: filehandlesall where 0 < count each {-21!x} each filehandlesall;
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `where 0<count each {-21!x} each filehandlesall` used twice, could define as a temp var on the first line and use it in the second.

File: code/processes/housekeeping.q [hunk 5]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
...
+
+	CompressedTable::(flip enlist[`files]!enlist`$compressedfiles)!{-21!x} each compressedfilehandles;
+	-1"Compression information for each file:";
+	show CompressedTable
+
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: Remove this line.

File: code/processes/housekeeping.q [hunk 6]
```diff
@@ -60,6 +73,18 @@ wrapper:{[DICT]
...
 //FUNCTIONS FOR LINUX
 
+
+
+//-compress by calling .cmp.compress function defined
+kdbzip:{[FILE]   
+	 @[{.lg.o[`housekeeping;"compressing ",x]; .cmp.compress[hsym `$x ;2;17;4;hcount hsym `$x]};
+	     FILE; 
+		 {.lg.e[`housekeeping;"Failed to compress ",x," : ", y]}[FILE]] 
+	  }
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `};`

File: code/processes/housekeeping.q [hunk 7]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
...
+	files: asc last each ("/" vs' hdbfilepaths);
+    filehandlesall: asc hsym `$hdbfilepaths;
+	compressedfiles: files where 0 < count each {-21!x} each filehandlesall;
+	compressedfilehandles: filehandlesall where 0 < count each {-21!x} each filehandlesall;
+
+	CompressedTable::(flip enlist[`files]!enlist`$compressedfiles)!{-21!x} each compressedfilehandles;
+	-1"Compression information for each file:";
+	show CompressedTable
+
+        }
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `};`

File: code/processes/housekeeping.q [hunk 8]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
...
+			wrapper each housekeepingcsv2]];
+		
+	 //-list all the compressed file and save in a table
+	hdbfilepaths: filepaths where filepaths like "*database20*";
+	files: asc last each ("/" vs' hdbfilepaths);
+    filehandlesall: asc hsym `$hdbfilepaths;
+	compressedfiles: files where 0 < count each {-21!x} each filehandlesall;
+	compressedfilehandles: filehandlesall where 0 < count each {-21!x} each filehandlesall;
+
+	CompressedTable::(flip enlist[`files]!enlist`$compressedfiles)!{-21!x} each compressedfilehandles;
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: Can you avoid the global amend `::`?

File: code/processes/housekeeping.q [hunk 9]
```diff
@@ -50,7 +50,20 @@ csvloader:{[CSV]
...
+	-1"Compression information for each file:";
+	show CompressedTable
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `:show CompressedTable;`

File: code/processes/housekeeping.q (hunk 8 above)
Comment (Kinoye) [reply code/processes/housekeeping.q]: Thanks, I would like to store that table globally, so one can retrieve anytime

File: code/processes/housekeeping.q (hunk 4 above)
Comment (Kinoye) [reply code/processes/housekeeping.q]: Good idea

File: code/processes/housekeeping.q (hunk 8 above)
Comment (ThomasSmyth) [reply code/processes/housekeeping.q]: Use `set` or put it in a namespace to make it global.

File: code/processes/housekeeping.q (hunk 9 above)
Comment (Kinoye) [reply code/processes/housekeeping.q]: What does this do?

File: code/processes/housekeeping.q (hunk 8 above)
Comment (Kinoye) [reply code/processes/housekeeping.q]: Do you mean using set to save on disk?

File: code/processes/housekeeping.q [hunk 10]
```diff
@@ -58,8 +58,18 @@ wrapper:{[DICT]
...
 
-//FUNCTIONS FOR LINUX
 
+
+//compress by calling .cmp.compress function defined using -19!
+kdbzip:{[FILE]   
+	 @[{.lg.o[`housekeeping;"compressing ",x]; .cmp.compress[filehandles;2;17;4;hcount filehandles:hsym `$x]};
+	     FILE; 
+	 {.lg.e[`housekeeping;"Failed to compress ",x," : ", y]}[FILE]] 
+	  }
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `};`

File: code/processes/housekeeping.q [hunk 11]
```diff
@@ -58,8 +58,18 @@ wrapper:{[DICT]
...
 	(value DICT[`function]) each (find[.rmvr.removeenvvar [DICT[`path]];DICT[`match];DICT[`age];DICT[`agemin]] except find[.rmvr.removeenvvar [DICT[`path]];DICT[`exclude];DICT[`age];DICT[`agemin]])]}
 
-//FUNCTIONS FOR LINUX
 
+
+//compress by calling .cmp.compress function defined using -19!
+kdbzip:{[FILE]   
+	 @[{.lg.o[`housekeeping;"compressing ",x]; .cmp.compress[filehandles;2;17;4;hcount filehandles:hsym `$x]};
+	     FILE; 
+	 {.lg.e[`housekeeping;"Failed to compress ",x," : ", y]}[FILE]]
```
Comment (ThomasSmyth) [inline code/processes/housekeeping.q]: `... x," : ",y]}FILE]`

**Pull request merged**